nautobot/dcim/filters/__init__.py

```diff
@@ -1008,12 +1013,13 @@ class InterfaceFilterSet(
     StatusModelFilterSetMixin,
     RoleModelFilterSetMixin,
 ):
-    # Override device and device_id filters from DeviceComponentModelFilterSetMixin to
+    # Override device and device_id filters from ModularDeviceComponentModelFilterSetMixin to
     # match against any peer virtual chassis members
-    device = MultiValueCharFilter(
+    device = NaturalKeyOrPKMultipleChoiceFilter(
+        queryset=Device.objects.all(),
+        to_field_name="name",
+        label="Device (name or ID)",
```

Comment:
Can/should we mark the separate `device_id` filter as deprecated now?

Reply:
We just mark deprecated filters with a todo comment, right?

Reply:
TODO plus updating its `label`, I believe.